Meson implementation

Totto16 opened the pull request:
This adds a meson build system to the project

- Some of the real benefits over CMake is e.g. the generation of the `src/parser_nodes` files.
Meson recompiles them on demand (like make but for the `main.py` and `parser_nodes.types`) , when you also recompile the other things, can CMake do that to :thinking:  

- It also adds a matrix build for windows and Linux with meson CI

- It removes both files (from VCS) in `src/parser_nodes` since they are just generated outputs, so also the CMake CI builds them, when they're needed.

- Since the two files are missing, the IDE probably complain, so meson gets a run command (run with "meson compile generate_parser_nodes" (add -C \<builddir\> if necessary)

- Meson doesn't support in source file generation, so no automatic rebuild for the src directory yet :sweat: and no automatic  generating for CMake :sweat: 

- [x] adapted to changes from #8 

Feel free to suggest changes :heart:

Totto16 committed:
ported to meson

Totto16 committed:
imrpoved the vcpkg integration, now no wraps are needed, only the vcpkg manifest (still WIP on windows)

Totto16 committed:
added grabber.py, and refactored it, so that both windows and linux use it

Totto16 committed:
- removed grabber.sh
- refactored vcpkg into subdir, now it's hidden in the main file
- added meson CI (matrix build for windows + linux)

Totto16 committed:
fixed some minor errors and made some minor improovements
- tested i with msvc 14.35.32215
- since it's using -Wall (/W4) and -Werror (/Wx) a minor error had to be muted in a library 😓

Totto16 committed:
added fetching submodules, otherwise it fails

Totto16 committed:
fixed macro

Totto16 committed:
try to add cmake parser_nodes support

Totto16 committed:
fixed cmake python generator code

Totto16 committed:
added env so that the windows vm can find the modules (locally it worked without it)

Totto16 committed:
meson tests for getting the generated files in both directories

Totto16 committed:
remove env, it's not needed anymore (see issue in submodule)

Totto16 committed:
added python to have newer version of python (3.9 fails, only on windows 3.9 is the default)

mgerhold commented:
Thanks for the huge amount of work you've put into this! 🙏
Please give me a few days to get back to this, since I have no capacity to look at it at the moment. Sorry for that!

Totto16 commented:
This has 0 priority, so no need to hurry :wink:

Totto16 committed:
refactored meson + vcpkg integration
the reason for this change is, that previously every dependency in the vcpkg.json was added unconditionally to a predefined var (deps), that is really bad, so now it adds the libraries it find in the vcpkg.json and then you can decide where you want to use these installed libraries, so you can use some in some targets and don#t have to use all in all targets

Totto16 committed:
adjusted the generation of parser_node.types

Totto16 committed:
adjusted the chnages to the newest status of main

Totto16 committed:
added header for std::unreachable

Totto16 committed:
fix environment errors

Totto16 commented:
It is now ready, and the CI can be invoked @mgerhold
